dapps/src/lib.rs

```diff
@@ -266,7 +266,7 @@ impl Server {
 	#[cfg(test)]
 	/// Returns address that this server is bound to.
 	pub fn addr(&self) -> &SocketAddr {
-		self.server.as_ref().expect("server is always Some at the start; it's consumed only when object is dropped; qed").addr()
+		&self.server.as_ref().expect("server is always Some at the start; it's consumed only when object is dropped; qed").addrs()[0]
```

Comment:
panicker: prove or remove!

what guarantee is that that there is at least one address in `addrs`? in any case better to use `.front()` to make this explicit.